src/sentry/services/hybrid_cloud/integration/service.py

```diff
@@ -143,6 +144,30 @@ def get_organization_context(
         by either integration_id, or a combination of provider and external_id.
         """
 
+    def get_organization_context__tmp(
+        self,
+        *,
+        organization_id: int,
+        integration_id: int | None = None,
+        provider: str | None = None,
+        external_id: str | None = None,
+    ) -> tuple[RpcIntegration | None, RpcOrganizationIntegration | None]:
```

Comment:
Should this method have the same signature as `get_organization_context__tmp` has in the impl module? The method defined in impl has a return of `RpcOrganizationIntegrationContextResult`

Reply:
This is confusing because of the single/plural distinction between `get_organization_context` and `get_organization_contexts`. There is no corresponding impl for `get_organization_context__tmp`. Note also that `RpcOrganizationIntegrationContextResult` replaces a tuple that has a *list* of `RpcOrganizationIntegration` as the second element, whereas `get_organization_context` returns only a single nullable `RpcOrganizationIntegration`.

My goal with `get_organization_context` (singular) is to remove it as an `@rpc_method`. It's a convenience method that delegates to `get_organization_contexts` and unpacks a singleton result. That unpacking step currently happens remotely, but we can do it locally instead by pulling it up into the service class as a concrete method. The change has no performance benefit (it's a single round trip either way) but lets us delete `get_organization_context` from the RPC API

Reply:
Ah bugger I missed the `s` when looking at the method names.